docs/api.rst

```diff
@@ -64,6 +64,17 @@ Currently supported data loader types are :py:class:`DictDataLoader`,
 :py:class:`NestedDictDataLoader`, and :py:class:`GFDLDataLoader` Each of these inherit
 from the abstract base :py:class:`DataLoader` class.
 
+.. note::
+
+   An important consideration can be the datatype used to store values in
+   one's datasets.  In particular, if the float32 datatype is used in
+   storage, it can lead to undesired inaccuracies in the computation of
+   reduction operations (like means) due to upstream issues.  To address
+   this it is recommended to always upcast float32 data to float64.  This
+   behavior is turned on by default.  If one would like to disable this
+   behavior one can set the ``upcast_float32`` argument in their ``DataLoader``
+   constructors to ``False``.
+
```

Comment:
Let's add a link to the xarray issue in these notes, as a parenthetical after "upstream issues", as you did in the What's New.

Also, let's switch these to be the second person imperative, e.g. 'your dataset', and 'You can disable this behavior by setting the ...'.  That makes it feel more inviting.